text/src/autogluon/text/automm/models/numerical_transformer.py

```diff
@@ -0,0 +1,279 @@
+        Parameters
+        ----------
+        prefix
+            The model prefix.
+        in_features
+            Dimension of input features.
+        d_token
+            The size of one token for `_CategoricalFeatureTokenizer`.
+        cls_token
+            If `True`, cls token will be added to the token embeddings.
+        out_features
+            Dimension of output features.
+        num_classes
+            Number of classes. 1 for a regression task.
+        token_bias
+            If `True`, for each feature, an additional trainable vector will be added in `_CategoricalFeatureTokenizer` 
+            to the embedding regardless of feature value. Notablly, the bias are not shared between features.
+        token_initialization
+            Initialization policy for parameters in `_CategoricalFeatureTokenizer` and `_CLSToke`. 
+            Must be one of `['uniform', 'normal']`. 
+        n_blocks
+            Number of the `FT_Transformer` blocks, which should be non-negative.
+        attention_n_heads
+            Number of attention heads in each `FT_Transformer` block, which should be postive.
+        attention_initialization
+            Weights initalization scheme for Multi Headed Attention module.
+        attention_dropout
+            Dropout ratio for the Multi Headed Attention module.
+        residual_dropout
+            Dropout ratio for the linear layers in FT_Transformer block.
+        ffn_activation
+            Activation function type for the Feed-Forward Network module.
+        ffn_normalization
+            Normalization scheme of the Feed-Forward Network module.
+        ffn_d_hidden
+            Number of the hidden nodes of the linaer layers in the Feed-Forward Network module.
+        ffn_dropout
+            Dropout ratio of the hidden nodes of the linaer layers in the Feed-Forward Network module.
+        prenormalization, first_prenormalization
+            Prenormalization to stablize the training.
+        kv_compression_ratio
+            The compression ration to reduce the input sequence length.
+        kv_compression_sharing
+            If `true` the projections will share weights.
+        head_activation
+            Activation function type of the MLP layer.
+        head_normalization
+            Normalization scheme of the MLP layer.
+
+        References
+        ----------
+        Yury Gorishniy, Ivan Rubachev, Valentin Khrulkov, Artem Babenko, 
+        "Revisiting Deep Learning Models for Tabular Data", 2021
+        https://arxiv.org/pdf/2106.11959.pdf
+        """
+
+        super().__init__()
+
+        assert d_token > 0, 'd_token must be positive'
+        assert n_blocks >= 0, 'n_blocks must be non-negative' 
+        assert attention_n_heads>0, 'attention_n_heads must be postive'
+        assert token_initialization in ['uniform', 'normal'], 'initialization must be uniform or normal'
+
+        self.prefix = prefix
+        self.numerical_key = f"{prefix}_{NUMERICAL}"
```

Comment:
We need to use property to set these keys. Refer to the latest code in other model classes.

Reply:
Consider change it to:

```python
@property
def numerical_key(self):
   return f"{prefix}_{NUMERICAL}"
```